src/main/kotlin/org/hyacinthbots/lilybot/extensions/util/ModUtilities.kt

```diff
@@ -418,6 +444,107 @@ class ModUtilities : Extension() {
 			}
 		}
 
+		unsafeSlashCommand {
+			name = "reset"
+			description = "'Resets' Lily for this guild by deleting all database information relating to this guild"
+
+			initialResponse = InitialSlashCommandResponse.None
+
+			requirePermission(Permission.Administrator) // Hide this command from non-administrators
+
+			check {
+				anyGuild()
+				hasPermission(Permission.Administrator)
+			}
+
+			action {
+				val modal = event.interaction.modal("Reset data for this guild", "resetModal") {
+					actionRow {
+						textInput(TextInputStyle.Short, "confirmation", "Confirm reset") {
+							placeholder = "Type 'yes' to confirm"
+						}
+					}
+				}
+
+				val interaction =
+					modal.kord.waitFor<ModalSubmitInteractionCreateEvent>(120.seconds.inWholeMilliseconds) {
+						interaction.modalId == "resetModal"
+					}?.interaction
+
+				if (interaction == null) {
+					modal.createEphemeralFollowup { content = "Reset interaction timed out" }
+					return@action
+				}
+
+				val confirmation = interaction.textInputs["confirmation"]!!.value!!
+				val modalResponse = interaction.deferEphemeralResponse()
+
+				if (confirmation.lowercase() != "yes") {
+					modalResponse.respond { content = "Confirmation failure. Reset cancelled" }
+					return@action
+				}
+
+				var response: EphemeralMessageInteractionResponse? = null
+
+				response = modalResponse.respond {
+					content = "Are you sure you want to reset the database? This action is **irreversible**."
+
+					components {
+						ephemeralButton(0) {
+							label = "I'm sure"
+							style = ButtonStyle.Danger
+
+							action {
+								response?.edit {
+									content = "Database reset!"
+									components { removeAll() }
+								}
+
+								guild?.getChannelOf<GuildMessageChannel>(
+									ModerationConfigCollection().getConfig(guild!!.id)?.channel ?: guild!!.asGuild()
+										.getSystemChannel()!!.id
+								)?.createMessage {
+									embed {
+										title = "Database Reset!"
+										description = "All data associated with this guild has been reset."
+										timestamp = Clock.System.now()
+										color = DISCORD_BLACK
+									}
+								}
+
+								// Reset
+								LoggingConfigCollection().clearConfig(guild!!.id)
+								ModerationConfigCollection().clearConfig(guild!!.id)
+								SupportConfigCollection().clearConfig(guild!!.id)
+								UtilityConfigCollection().clearConfig(guild!!.id)
+								GalleryChannelCollection().removeAll(guild!!.id)
+								LogUploadingBlacklistCollection().clearBlacklist(guild!!.id)
+								RoleMenuCollection().removeAllRoleMenus(guild!!.id)
+								TagsCollection().clearTags(guild!!.id)
+								for (it in ThreadsCollection().getAllThreads()) {
```

Comment:
Okay we really should migrate thread data to have guildid

Reply:
I'm literally so happy you said this :D